src/gui/guiText.cpp

```diff
@@ -0,0 +1,1000 @@
+
+	} else if (name == "img" || name == "item") {
+		if (end)
+			return 0;
+
+		// Required attributes
+		if (!attrs.count("name"))
+			return 0;
+
+		newTag(name, attrs);
+
+		if (name == "img")
+			enterElement(ELEMENT_IMAGE);
+		else
+			enterElement(ELEMENT_ITEM);
+
+		m_element->text = strtostrw(attrs["name"]);
+
+		if (attrs.count("float")) {
+			if (attrs["float"] == "left")
+				m_element->floating = FLOAT_LEFT;
+			if (attrs["float"] == "right")
+				m_element->floating = FLOAT_RIGHT;
+		}
+
+		if (attrs.count("rotate") && attrs["rotate"] == "yes")
+			m_element->rotation = IT_ROT_OTHER;
+
+		if (attrs.count("width")) {
+			int width = strtol(attrs["width"].c_str(), NULL, 10);
+			if (width > 0)
+				m_element->dim.Width = width;
+		}
+
+		if (attrs.count("height")) {
+			int height = strtol(attrs["height"].c_str(), NULL, 10);
+			if (height > 0)
+				m_element->dim.Height = height;
+		}
+		endElement();
+
+	} else if (name == "link") {
+		if (end)
+			closeTag(name);
+		else {
+			if (!attrs.count("name"))
+				return 0;
+			openTag(name, attrs)->style["color"] = m_style["linkcolor"];
+		}
+	} else if (name == "center" || name == "justify" || name == "left" || name == "right") {
+		if (end)
+			closeTag(name);
+		else
+			openTag(name, attrs)->style["halign"] = name;
+		endParagraph();
+	} else if (name == "normal") {
+		if (end)
+			closeTag(name);
+		else
+			openTag(name, attrs)->style["fontsize"] = "16";
+		endElement();
+	} else if (name == "big") {
+		if (end)
+			closeTag(name);
+		else
+			openTag(name, attrs)->style["fontsize"] = "24";
+		endElement();
+	} else if (name == "bigger") {
+		if (end)
+			closeTag(name);
+		else
+			openTag(name, attrs)->style["fontsize"] = "36";
+		endElement();
+	} else if (name == "style") {
+		if (end)
+			closeTag(name);
+		else {
+			if (attrs.count("color") && check_color(attrs["color"]))
+				style["color"] = attrs["color"];
+
+			if (attrs.count("font") &&
+				(attrs["font"] == "mono" || attrs["font"] == "normal"))
+				style["fontstyle"] = attrs["font"];
+
+			if (attrs.count("size") && strtol(attrs["size"].c_str(), NULL, 10) > 0)
+				style["fontsize"] = attrs["size"];
+
+			openTag(name, attrs)->style = style;
+		}
+		endElement();
+	} else return 0; // Unknown tag
+
+	// Update styles accordingly
+	m_style.clear();
+ 	for (auto tag = m_active_tags.crbegin(); tag != m_active_tags.crend(); ++tag)
+		for (auto const& prop : (*tag)->style)
+			m_style[prop.first] = prop.second;
+
+	return cursor;
+}
+
+// -----------------------------------------------------------------------------
+// Text Drawer
+
+TextDrawer::TextDrawer(
+	const wchar_t* text,
+	Client* client,
+	gui::IGUIEnvironment* environment,
+	ISimpleTextureSource *tsrc):
+	m_text(text),
+	m_client(client),
+	m_environment(environment)
+{
+	// Size all elements
+	for (auto & p : m_text.m_paragraphs) {
+		for (auto & e : p.elements) {
+			if (e.type == ParsedText::ELEMENT_SEPARATOR ||
+				e.type == ParsedText::ELEMENT_TEXT)
+			{
```

Comment:
multiple code style issues

Reply:
i.e. double-indents for continuation lines and opening brace on the same line

```cpp
if (e.type == ParsedText::ELEMENT_SEPARATOR ||
		e.type == ParsedText::ELEMENT_TEXT) {
```